### PR TITLE
db connection and sockets setup

### PR DESCRIPTION
**Pull request by ali-iqbal-brainx**
1. chat routes
2. users routes
3. auth routes
4. db setup
5. socket connection

### CODE REVIEW

File: controllers/chatController.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+        if (!name) {
+            return response
+                .status(406)
+                .json({
+                    error: "Required fields are missing"
+                });
+        }
+
+        const chatCode = Math.floor(100000 + Math.random() * 900000).toString();
```
Comment (ali-faiz-brainx) [inline controllers/chatController.js]: use constant values instead of hardcoded

File: index.js [hunk 2]
```diff
@@ -1,10 +1,24 @@
...
 const bodyParser = require('body-parser');
-const port = 4000;
+app.use(bodyParser.urlencoded({ limit: '50mb', extended: true }));
+const port = process.env.PORT || 4000;
 
 
+
+app.use(express.json({ limit: '50mb' }));
+const corsOrigin ={
+    origin:'http://localhost:3000', //or whatever port your frontend is using
```
Comment (ali-faiz-brainx) [inline index.js]: Pust constents in envs instead of hard coded

**Review by ali-faiz-brainx — CHANGES_REQUESTED** [review]

File: controllers/authController.js [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+        console.log(error);
+        response.status(500).json({
+            error: "Something went wrong",
+        });
+    }
+}
+
+const signup = async (request, response) => {
+    try {
+        const { name, password } = request.body;
```
Comment (usama-masood-brainx) [inline controllers/authController.js:79]: check if user already exist

File: index.js [hunk 4]
```diff
@@ -1,13 +1,79 @@
...
+const { Server } = require("socket.io");
 const bodyParser = require('body-parser');
-const port = 4000;
+app.use(bodyParser.urlencoded({ limit: process.env.EXPRESS_LIMIT, extended: true }));
+const db = require('./configs/mongoDb');
+const port = process.env.PORT || 4000;
 
+app.use(express.json({ limit: process.env.EXPRESS_LIMIT }));
+const corsOrigin = {
+    origin: 'http://localhost:3000', //or whatever port your frontend is using
```
Comment (usama-masood-brainx) [inline index.js]: put dynamic constants in env file

File: index.js [hunk 5]
```diff
@@ -1,13 +1,79 @@
...
+}
+app.use(cors(corsOrigin));
 
-app.listen(port, () => {
-    console.log(`app listening on port ${port}`);
+const server = http.createServer(app);
+
+const io = new Server(server, {
+    cors: {
+        origin: 'http://localhost:3000',
```
Comment (usama-masood-brainx) [inline index.js]: put dynamic constants in env file

File: middlewares/userMiddleware.js [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+
+        if (!name || !password) {
+            return response
+                .status(406)
+                .json({
+                    error: "Required fields are missing"
+                });
+        }
+
+        if (password.length<8) {
```
Comment (usama-masood-brainx) [inline middlewares/userMiddleware.js]: use password regex for the pattern (1 uppercase, 1 symbol, 1 number, and min 8 char)

File: models/one_to_one.js [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+                type: ObjectId,
+                require: true,
+                default: new ObjectId()
+            },
+            userId: {
+                type: ObjectId,
+                ref: "users",
+                require: true
+            },
+            seenBy: [],
```
Comment (usama-masood-brainx) [inline models/one_to_one.js:19]: define type of object in seenBy array

File: services/jwtService.js [hunk 8]
```diff
@@ -0,0 +1,22 @@
+require('dotenv').config();
```
Comment (usama-masood-brainx) [inline services/jwtService.js]: require dotenv at start of server script instead of individual modules

**Review by usama-masood-brainx — CHANGES_REQUESTED** [review]

**Pull request merged**